Images/ImageImportExport/ImageImport.cpp

```diff
@@ -0,0 +1,141 @@
+		std::cout << message << std::endl;
+		ASfree((void*)iName);
+		ASfree((void*)oName);
+
+        // Convert the input file into a PDEImage.
+		PDEImage newImage = DLCreatePDEImageFromFile(doc, sInput, NULL);
+
+		/* 
+			Create a PDF page which is one inch larger all around this image.
+			The design width and height for the image are carried in the
+			matrix.a and matrix.d fields, respectively.
+			There are 72 PDF user space units in one inch.
+		*/
+		ASDoubleMatrix matrix = { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 };
+		PDEElementGetMatrixEx((PDEElement)newImage, &matrix);
+
+		ASFixedRect bbox;
+		bbox.left = 0;
+		bbox.top = FloatToASFixed(matrix.d + 144);
+		bbox.right = FloatToASFixed(matrix.a + 144);
+		bbox.bottom = 0;
+
+		PDPage page = PDDocCreatePage(doc, PDDocGetNumPages(doc) - 1, bbox);
+
+		// Center the image in its bounding box.
+		DLPDEImageTranslate(newImage, 72, 72);
+
+		// Manimulate the image by rotating and scaling it.
+		DLPDEImageRotate(newImage, 180);
+		DLPDEImageScale(newImage, 0.5, 0.5);
+
+		// Adjust the image location on the page to have more space for the next steps
+		DLPDEImageTranslate(newImage, DLPDEImageGetWidth(newImage) * -1, DLPDEImageGetHeight(newImage) * -1.25);
+
+		// Make a clone of the original then manipulate it on the page.
+		PDEElement copyElement = PDEElementCopy(reinterpret_cast<PDEElement>(newImage), 0);
+		PDEImage clone = reinterpret_cast<PDEImage>(copyElement);
+
+		DLPDEImageTranslate(clone, 72, 72);
+		DLPDEImageRotate(clone, 90);
+		DLPDEImageScale(clone, 0.5, 0.5);
+
+		/* 
+			Use the cloned image to create a new grayscale image then manipulate it on the page.
+			DLColorConvertPDEImage uses an AC_ProfileCode value: AC_Profile_DotGain30 and an AC_RenderIntent value: AC_Saturation
+			to create a grayscale copy of the image.
+		*/
+		PDEImage grayscaleImage = DLColorConvertPDEImage(&doc, clone, AC_Profile_DotGain30, AC_Saturation, true);	
+		DLPDEImageTranslate(grayscaleImage, -72, -72);
+		DLPDEImageRotate(grayscaleImage, 180);
+		DLPDEImageScale(grayscaleImage, 0.5, 0.5);
+
+		/* 
+			The DLPDEImageExportParams struct is mainly used when exporting images, but it is also required for DLCreateResampledPDEImage().
+			This is because DLCreateResampledPDEImage() uses exportParams->ExportDPI in its calculations and apdates it accordingly.
+			DLCreateResampledPDEImage() creates a copy of the original PDEImage with the specified values.
+		*/
+		DLPDEImageExportParams exportParams = DLPDEImageGetExportParams();
+		PDEImage newRezImage = DLCreateResampledPDEImage(clone, &exportParams, 4);
+		DLPDEImageTranslate(newRezImage, -172, 72);
+		DLPDEImageRotate(newRezImage, 270);	
+		
+		/*
+			Some PDEImage's contain soft masks which can be retrived using DLPDEImageGetSoftMask(&image). 
+			The soft mask of a PDEImage can also be set using the fucntion below. 
+			If the function is passed a nullptr the current soft mask is deleted. 
```

Comment:
let's remove the comment, the sample shouldn't be used as a form of documentation